### ISSUE TITLE
reviver for toJS

### ISSUE THREAD

**Issue by jardakotesovec** [facebook/immutable-js#388]
I am using custom reviver with `fromJS`, which detects arrays with object having `id` property which I turn into `OrderedMap` instead of default `List`, which is very useful for traversing large JSON structures. Now I need to make the conversion other way round (OrderedMap -> array), but `toJS` does not support `reviver`. 

Could be this considered as feature request? Or is there trick I can use?

Thanks!

**Comment by wmertens** [comment]
you could perhaps override `.toJS()`?

**Comment by leebyron** [comment]
Using reviver here is interesting. Could you illustrate the API you expected in order to achieve what you're trying to achieve?

**Comment by jardakotesovec** [comment]
```
myMap.toJS(function (key, value) {
  // I assume that first() would return converted JS type ?
  if (Immutable.OrderedMap.isOrderedMap(value) && value.size && value.first().id) { 
    return value.toArray()
  }
  // trying to mimic the default behaviour
  else if (Immutable.Iterable.isIndexed(value)) {
   return value.toObject()
  }
  else {
    return value.toArray() 
  }
})
```

Does it make sense? 

For reference - [here](https://github.com/facebook/immutable-js/issues/57#issuecomment-77529149) is the reviver I use with fromJS. It also took some elaborating to make it work as dealing with seq coming from reviver was kind of magic - some api seq calls end up with undefined errors, but getting value via `getIn` worked eventually. I think it have something to do with seq lazy behavior..

**Comment by facebook-github-bot-4** [comment]
Thank you for reporting this issue and appreciate your patience. We've  notified the core team for an update on this issue. We're looking for a response within the next 30 days or the issue may be closed.

**Comment by shaharyakir** [comment]
Any news on this?

**Comment by leebyron** [comment]
Up for grabs to ambitious contributors.

In my opinion `toJS` is one of the most abused parts of the API - so perhaps this is a good opportunity to break from the norm and implement `Immutable.toJS()` instead of adding weight to the class method.

**Comment by thecrogers** [comment]
@leebyron I agree that adding ```toJS`` to  ```Immutable.toJS()``` would be a good direction to go with this request. Would we want to remove the existing ```Collection.toJS()```, or share the new functionality in ```Immutable.toJS()```.
